components/nimbus/src/error.rs

```diff
@@ -63,6 +64,8 @@ pub enum NimbusError {
     ParseIntError(#[from] ParseIntError),
     #[error("Transform parameter error: {0}")]
     TransformParameterError(String),
+    #[error("Error with HTTP client: {0}")]
+    ClientError(#[from] rs_client::ClientError),
```

Comment:
I'm curious whether there's any desire for this mapping to be more granular, exposing more detailed errors from the shared component, and what the best way to do that would be.

Reply:
It looks like this already does that - this enum variant could be unpacked as `NimbusError::ClientError(internal)` where `internal` will itself be a `rs_client::error::ClientError` enum - so code that cares could do something like:

```rust
match err {
  NimbusError::ClientError(rs_client::error::ClientError::RequestError(_)) -> println!(...),
  _ -> println!(...),
}
```
if it really wanted to know about just that one error, or could do:
```rust
match err {
  NimbusError::ClientError(_) -> println!(...),
  _ -> println!(...),
}
```
if it wanted to handle all `ClientError`s together.

However, that means that some of the other errors here are redundant - eg, it seems likely that `BackoffError` etc should be removed from this enum as they are now sub-variants of `ClientError`.